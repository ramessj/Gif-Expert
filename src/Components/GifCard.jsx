import { useState } from "react"

const GifCard = ({title, url}) => {

    const [titleName, setTitleName] = useState(title);

    const checkTitle = () => {
        if(titleName == ''){
            setTitleName('GIF');
        }
    }

    checkTitle();


    return (
        <div className="card">
            <img src={url} alt={title} />
            <p>{titleName}</p>
        </div>
    )
}

export default GifCard