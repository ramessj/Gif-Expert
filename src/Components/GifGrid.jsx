import { useEffect } from "react";
import { getGifs } from "../Helpers/getGifs";

export const GifGrid = ({category}) => {

    

    useEffect( () => {
        getGifs({category});

    }, []);



    return (
        <>
            <h3>{category}</h3>
            
            
        </>
)};

