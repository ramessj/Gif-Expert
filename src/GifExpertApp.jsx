import { useState } from 'react';
import { AddCategory, GifGrid } from './Components';


export const GifExpertApp = () => {
	const [ categories, setCategories ] = useState( [ 'GIFS' ] );

	const onAddCategory = ( newCategory ) => {
		const currentCategoriesUpperCase = categories.map( ( cat ) => 
			cat.toUpperCase()
		);

		if ( currentCategoriesUpperCase.includes( newCategory.toUpperCase() ) ) return;

		function toUpper( str ) {
			return str
				.toLowerCase()
				.split( ' ' )
				.map( function ( word ) {
					return word[0].toUpperCase() + word.substr( 1 );
				} )
				.join( ' ' );
		}

		setCategories( [ toUpper( newCategory ), ...categories ] );
	};

	return (
		<>
			<h2>GIF EXPERT</h2>

			<AddCategory onNewCategory={( e ) => onAddCategory( e )} />

			{categories.map( ( category ) => (
				<GifGrid key={category} category={category} />
			) ) }
		</>
	);
};
