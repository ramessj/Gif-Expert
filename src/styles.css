@import url('https://fonts.googleapis.com/css2?family=Delicious+Handrawn&display=swap');

html {
  background-color: rgb(234, 234, 234);  
}

body {
  padding: 2rem 1rem;
  font-family: Helvetica, Arial, sans-serif;
  background-color: rgb(234, 234, 234);
}

#root {
  display: flex;
  flex-direction: column;
  align-items: center;
}

h2 {
  font-family: 'Delicious Handrawn', cursive;

  font-size: 3rem;
  margin-bottom: 2rem;
  color: rgb(97, 32, 158);
}

form {
  margin-bottom: 2rem;
  width: 100%;
  display: flex;
  justify-content: center;
}

input {
  background-color: white;
  border-radius: 5px;
  border: 1px solid rgb(97, 32, 158);
  color: black;
  font-size: 1.2rem;
  outline: none;
  padding: 10px 15px;
  width: 90%;
}



h3 {
  font-size: 1.5rem;
  margin-bottom: 2rem;
  text-align: center;
  color: rgb(92, 92, 92);
  position: relative;
}

h3::after {
  content: "";
  position: absolute;
  bottom: -0.5rem;
  left: 0;
  width: 100%;
  border: 1px solid rgb(97, 32, 158);
  border-radius: 1rem;
  -webkit-border-radius: 1rem;
  -moz-border-radius: 1rem;
  -ms-border-radius: 1rem;
  -o-border-radius: 1rem;
}

.cardsContainer {
  width: 100%;
  padding-top: 1.5rem;
  border-radius: 1rem;
  border-top: 1px solid rgb(97, 32, 158);
  -webkit-border-radius: 1rem;
  -moz-border-radius: 1rem;
  -ms-border-radius: 1rem;
  -o-border-radius: 1rem;
}

.cardsTitle {
  display: flex;
  justify-content: center;
}

.card-grid {
  display: flex;
  flex-direction: row;
  flex-wrap: wrap;
  justify-content: center;
  gap: 2rem;
  margin-bottom: 2rem;
}

.card {
  align-content: center;
  align-items: center;
  background-color: white;
  border-radius: 10px;
  box-shadow: 0px 3px 5px rgba(0, 0, 0, 0.03);
  display: flex;
  flex-direction: column;
  height: 0%;
  justify-content: center;
  margin-bottom: 20px;
  overflow: hidden;
}

.card p {
  background-color: white;
  word-wrap: break-word;
  font-size: 1rem;
  padding: 1rem;
  text-align: center;
}

.card img {
  width: 25rem;
  max-width: 100%;
}


@media(min-width: 500px){
  
  h2{
    font-size: 4rem;
  }
  
  h3{
    font-size: 2.5rem;
    margin-bottom: 3rem;
  }
}