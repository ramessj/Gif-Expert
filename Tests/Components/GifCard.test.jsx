import { render, screen } from "@testing-library/react"
import  {GifCard}  from "../../src/Components"

describe('Pruebas en <GifCard />', () => {

  const title = 'prueba'

  const url = 'https://media1.giphy.com/dadada.gif'

  test('Debe hacer match con el snapshot', () => {
    const {container} = render(<GifCard title={title} url={url}/>)
    
    expect(container).toMatchSnapshot();

  })

  test('Debe mostrar la imagen con la URL y el ALT indicado', () => {

    render(<GifCard title={title} url={url} />)

    const { src, alt } = screen.getByRole('img');

    expect( src ).toBe( url )
    expect( alt ).toBe( title )

  })

  test('Debe de mostrar el titulo en el componente', () => {

    render(<GifCard title={title} url={url} />)

    expect( screen.getByText(title)).toBeTruthy()
  })




})