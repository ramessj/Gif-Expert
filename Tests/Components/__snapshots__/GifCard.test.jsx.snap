// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <GifCard /> Debe hacer match con el snapshot 1`] = `
<div>
  <div
    class="card"
  >
    <img
      alt="prueba"
      src="https://media1.giphy.com/dadada.gif"
    />
    <div
      class="cardSeparator"
    />
    <p>
      prueba
    </p>
  </div>
</div>
`;
